### PR TITLE
Switched to gRPC client load balancer and DNS resolver

### PR DESCRIPTION
**Pull request by yfei1**

### CODE REVIEW

**Comment by sawagh** [comment]
Have we considered using a service mesh? Here is an article explaining the advantages over the current approach:
https://kubernetes.io/blog/2018/11/07/grpc-load-balancing-on-kubernetes-without-tears/

The approach we take works for internal clients using our library - but how would this go for externally written clients? Linkerd seems to have a better latency based LB too - rather than just roundrobin. Thoughts?

File: .golangci.yaml [hunk 1]
```diff
@@ -181,6 +181,7 @@ linters:
     - interfacer # deprecated - "A tool that suggests interfaces is prone to bad suggestions"
     - gochecknoglobals
     - funlen
+    - gochecknoinits
```
Comment (jeremyje) [inline .golangci.yaml]: use `// nolint:gochecknoinits instead.`

File: install/helm/open-match/subcharts/open-match-customize/values.yaml [hunk 2]
```diff
@@ -18,10 +18,12 @@
 
 function:
   replicas: 3
+  portType: ClusterIP
   image: openmatch-mmf-go-soloduel
 
 evaluator:
   replicas: 3
+  portType: ClusterIP
```
Comment (jeremyje) [inline install/helm/open-match/subcharts/open-match-customize/values.yaml]: This is kinda weird to need here.

**Comment by yfei1** [comment]
> Have we considered using a service mesh? Here is an article explaining the advantages over the current approach:
> https://kubernetes.io/blog/2018/11/07/grpc-load-balancing-on-kubernetes-without-tears/
> 
> The approach we take works for internal clients using our library - but how would this go for externally written clients? Linkerd seems to have a better latency based LB too - rather than just roundrobin. Thoughts?

I think doing client side LB or Linked will both introduce an extra layer of complexity to Open Match and installing Linkerd for Open Match is also a non-trivial task. 
1. Linkerd itself includes Prometheus and Grafana installations, they claimed that their Prometheus is tightly bonded to Linkerd and is not optional - I think people may start to wonder why we need two different Prometheus for running Open Match if we choose this approach.
2. We may need to modify our PodSecurityPolicies to grant Linkered enough permission. I've tried working on installing it with Open Match on Friday but with no luck :(

**Comment by sawagh** [comment]
If the current approach works for now, I propose submitting it as a spot fix - and iterating over the issue still in v0.8 - I think all the concerns around this are valid but if a solution is ready and unblocks further scale testing, can we put it in?

**Comment by sawagh** [comment]
Remove this PR unless you plan to use it as reference as short term fix?

**Pull request closed without merging**